src/rich-log/log.hh

```diff
@@ -1,66 +1,90 @@
+ *
+ *    // per default, Trace is disabled compile-time
+ *    // a custom minimum compile-time verbosity level for your domain can be specified via
+ *    RICH_LOG_DECLARE_DOMAIN_MV(MyDomain, Warning);
+ */
 
-inline void warning_functor(MessageBuilder& builder)
-{
-    builder.set_severity(severity::warning());
-    builder.set_use_error_stream(true);
-}
+#define RICH_LOG_IMPL(Domain, Severity, Formatter, ...)                                                                                      \
+    do                                                                                                                                       \
+    {                                                                                                                                        \
+        if constexpr (rlog::verbosity::Severity >= rlog::domains::Domain::CompileTimeMinVerbosity)                                           \
+        {                                                                                                                                    \
+            if (rlog::verbosity::Severity >= rlog::domains::Domain::domain.min_verbosity)                                                    \
+            {                                                                                                                                \
+                static constexpr rlog::location _rlog_location = RLOG_LOCATION();                                                            \
+                if (rlog::detail::do_log(rlog::domains::Domain::domain, rlog::verbosity::Severity, &_rlog_location, Formatter(__VA_ARGS__))) \
+                    CC_DEBUG_BREAK();                                                                                                        \
+            }                                                                                                                                \
+        }                                                                                                                                    \
+    } while (0) // force ;
 
-inline void error_functor(MessageBuilder& builder)
-{
-    builder.set_severity(severity::error());
-    builder.set_use_error_stream(true);
-}
+/// writes an info log message to the Default domain using cc::format (printf AND pythonic syntax)
+#define RICH_LOG(...) RICH_LOG_IMPL(Default, Info, cc::format, __VA_ARGS__)
+/// writes an info log message to the Default domain using cc::formatp (pythonic  syntax only)
+#define RICH_LOGP(...) RICH_LOG_IMPL(Default, Info, cc::formatp, __VA_ARGS__)
+/// writes an info log message to the Default domain using cc::formatf (printf syntax only)
+#define RICH_LOGF(...) RICH_LOG_IMPL(Default, Info, cc::formatf, __VA_ARGS__)
+/// writes a log message with given domain and severity using cc::format
+#define RICH_LOGD(Domain, Severity, ...) RICH_LOG_IMPL(Domain, Severity, cc::format, __VA_ARGS__)
```

Comment:
well, we could just have the default be `cc::format` and then a different macro for a custom formatter. this would also tie into a different feature a have in mind: logging-specific formatters that add colors, metadata, etc. and produce a more structured string (richt text or json or something)